### ISSUE TITLE
Splunk Hec Exporter x509: certificate is not valid for any names, but wanted to match <Splunk_Link>

### ISSUE THREAD

**Issue by ericashi** [open-telemetry/opentelemetry-collector-contrib#24632]
### Component(s)

exporter/splunkhec

## Description
I am trying to export data into Splunk using splunkhecexporter with TLS insecure_skip_verify=false. When I tried to send it into Splunk, it will shows an error message. After that, I tried to add in the TLS certificate inside the configuration, it still shows the same error message. Please find the error message from the "Log Output" section. 

## Steps to Reproduce
1. Configure the .yaml configuration file
2. Setup a deployment using Opentelemetry image inside Rancher environment
3. Run the .yaml configuration file using the otelcol-contrib image in Rancher.
Please refer to the "OpenTelemetry Collector configuration" section for the configurations.

## Expected Result
I expected the data scraped from the Prometheus Server to be sent into Splunk.

## Actual Result
No data is sent into Splunk. "x509: certificate is not valid for any names" error is prompted. 
Please refer to "Log Output" for the full error message.

### Collector version

v0.75.0

## Environment
Rancher v2.6.11

### OpenTelemetry Collector configuration

```yaml
receivers:
  prometheus:
    config:
      scrape_configs:
      - job_name: 'node_exporter_filter_7'
        scrape_interval: 10s
        static_configs:
        - targets: ['<NODE EXPORTER ENDPOINT>']

processors:
  batch:

exporters:
  logging:
    verbosity: detailed

  splunk_hec:
    token: "SPLUNK HEC TOKEN"
    endpoint: "SPLUNK COLLECTOR URL"
    tls:
      ca_file: /etc/ssl/certs/cert.crt
      cert_file: /etc/ssl/certs/<Splunk_cert>.crt
      key_file: /etc/ssl/certs/<Splunk_Key>.key


service:
  telemetry:
    metrics:
      level: detailed
      address: 0.0.0.0:8885

  pipelines:
    metrics:
      receivers: [prometheus]
      processors: []
      exporters: [logging, splunk_hec]
```

### Log output

```shell
2023-07-26T06:12:01.308Z	info	exporterhelper/queued_retry.go:433	Exporting failed. Will retry the request after interval.	{"kind": "exporter", "data_type": "metrics", "name": "splunk_hec", "error": "Post \"https://<SPLUNK_LINK>:<SPLUNK_PORT>/services/collector\": x509: certificate is not valid for any names, but wanted to match <DNS NAME>", "interval": "7.402035261s"}
```

### Additional context

I have tried out for both binary and image with the same configurations. However, I got the same error when pushing into Splunk with or without configuring the TLS configurations.

May I get your assistance on the following questions please?
1. Is my configuration with the certs file correct?
a. If yes, how can I solve this issue?
b. If no, how should I configure it so that it will allow me to send data into Splunk?

**Comment by atoulme** [comment]
What is the DN represented by the cert? Can you see if the cert is valid? Is it expired? Is the domain name the proper domain name you expect?

**Comment by ericashi** [comment]
Hi @atoulme,

I am unable to provide the DN as it is private and confidential but the DN in the certificate is the SPLUNK_LINK without the port (https://<SPLUNK_LINK>:<SPLUNK_PORT>).
The cert should be valid as I got the certificates from the Splunk Team.
The cert is still usable and not expired.
Yes, the domain name is the proper domain name I expected.

With this information, may I know how should I proceed next?

Thank you.
